seahub/api2/views.py

```diff
@@ -3573,6 +3575,35 @@ def get(self, request, repo_id):
         except IOError as e:
             return api_error(status.HTTP_500_INTERNAL_SERVER_ERROR, str(e))
 
+_REPO_ID_PATTERN = re.compile(r'[-0-9a-f]{36}')
+
+class RepoTokensView(APIView):
+    authentication_classes = (TokenAuthentication, )
+    permission_classes = (IsAuthenticated,)
+    throttle_classes = (UserRateThrottle, )
+
+    @json_response
+    def get(self, request, format=None):
+        repos = request.GET.get('repos', None)
+        if not repos:
+            return api_error(status.HTTP_400_BAD_REQUEST, "You must specify libaries ids")
+
+        repos = [repo for repo in repos.split(',') if repo]
+        if any([not _REPO_ID_PATTERN.match(repo) for repo in repos]):
+            return api_error(status.HTTP_400_BAD_REQUEST, "Libraries ids are invalid")
+
+        if any([not seafile_api.check_repo_access_permission(
```

Comment:
What if a user no longer have a permission to a repo after logout? Then he will not be able to login again?